Import modules

In [8]:
import pandas as pd


What is the northernmost airport in the United States?

My guess is that the northernmost airport in the United States will be in Alaska, as it is the northernmost state.  Now we'll read and sort the data from .csv file posted on github and display the top 5 airports by latitude.

In [9]:
airports = pd.read_csv('https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/airports.csv')
airports.sort_values('lat', ascending=False).head(5)

faa                                        name        lat         lon  \
417  EEN                     Dillant Hopkins Airport  72.270833   42.898333   
230  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110  AIN                          Wainwright Airport  70.638056 -159.994722   
708  K03                               Wainwright As  70.613378 -159.860350   
152  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   

     alt   tz dst              tzone  
417  149 -5.0   A                 \N  
230   44 -9.0   A  America/Anchorage  
110   41 -9.0   A  America/Anchorage  
708   35 -9.0   A  America/Anchorage  
152   96 -9.0   A  America/Anchorage

The returned results tell us that Dillant Hopkins Airport is the northernmost based on the latitude of 72.270833.  Upon further inspection of the data, I noticed that the tzone field is inconsistent with the others.  I looked up the coordinates for each airport and discovered that the latitude and longitude coordinates for Dillant Hopkins Airport entered incorrectly.  The true northernmost airport is Wiley Post Will Rogers Memorial Airport.

What is the easternmost airport in the United States?

I believe that the easternmost airport will be in Maine, since no other state is further east.

Now we'll sort the data by longitude in order to figure out the easternmost airport.

In [10]:
airports = pd.read_csv('https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/airports.csv')
airports[(airports['tzone'].str.contains('America')==True)&(airports['tzone'].str.contains('Anchorage')==False)].sort_values('lon',ascending=[False]).head()

faa                                 name        lat        lon  alt  \
444   EPM           Eastport Municipal Airport  44.910111 -67.012694   45   
624   HUL                         Houlton Intl  46.123083 -67.792056  489   
259   CAR                         Caribou Muni  46.871500 -68.017917  626   
1101  PQI  Northern Maine Rgnl At Presque Isle  46.688958 -68.044797  534   
1398  WFK  Northern Aroostook Regional Airport  47.285556 -68.312778  988   

       tz dst             tzone  
444  -5.0   A  America/New_York  
624  -5.0   A  America/New_York  
259  -5.0   A  America/New_York  
1101 -5.0   A  America/New_York  
1398 -5.0   A  America/New_York

Seaching by just longitude values did not return the desired results.  The results received were for Alaska which is in the opposite direction that we are looking for.  In order to find the easternmost aiport, I had to remove the Anchorage Alaska results.  

Doing this returns Eastport Municipal Airport which is located in Maine, and is the easternmost aiport in the United Staes.

On February 12th, 2013, which New York area airport had the windiest weather?



In [13]:
weather = pd.read_csv('https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/weather.csv')
weather[(weather["month"]==2) & (weather['day']==12) & (weather['year']==2013)].sort_values('wind_speed', ascending=False).head(5)

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1015     EWR  2013      2   12     8  39.02  26.96  61.63     260.0   
18433    LGA  2013      2   12     7  42.98  26.06  50.94     290.0   
1024     EWR  2013      2   12    17  44.06  26.06  48.87     270.0   
18444    LGA  2013      2   12    18  44.06  23.00  43.02     300.0   
9728     JFK  2013      2   12    13  42.08  24.08  48.56     280.0   

       wind_speed    wind_gust  precip  pressure  visib             time_hour  
1015   1048.36058  1206.432388     0.0    1008.3   10.0  2013-02-12T08:00:00Z  
18433    23.01560    26.485892     0.0    1007.1   10.0  2013-02-12T07:00:00Z  
1024     21.86482    25.161598     0.0    1012.5   10.0  2013-02-12T17:00:00Z  
18444    21.86482    25.161598     0.0    1011.7   10.0  2013-02-12T18:00:00Z  
9728     20.71404    23.837303     0.0    1011.6   10.0  2013-02-12T13:00:00Z

The results show that EWR which is Newark Liberty International Aiport which is technically a New York Area airport.  If we are looking at strictly New York airports, then LaGuardia Aiport would be have had the windiest weather on February 12, 2013.